lib/cluster/cpg.c

```diff
@@ -398,51 +399,54 @@ pcmk__cpg_message_data(cpg_handle_t handle, uint32_t sender_id, uint32_t pid,
     }
 
     if (handle != 0) {
-        // Do filtering and field massaging
         uint32_t local_nodeid = pcmk__cpg_local_nodeid(handle);
         const char *local_name = pcmk__cluster_local_node_name();
 
-        if ((msg->sender.id != 0) && (msg->sender.id != sender_id)) {
-            crm_err("Nodeid mismatch from %" PRIu32 ".%" PRIu32
-                    ": claimed nodeid=%" PRIu32,
+        // Update or validate message sender ID
+        if (msg->sender.id == 0) {
+            msg->sender.id = sender_id;
+        } else if (msg->sender.id != sender_id) {
+            crm_warn("Ignoring CPG message from ID %" PRIu32 " PID %" PRIu32
+                    ": claimed ID %" PRIu32,
                     sender_id, pid, msg->sender.id);
             return NULL;
         }
+
+        // Ignore messages that aren't for the local node
         if ((msg->host.id != 0) && (local_nodeid != msg->host.id)) {
-            crm_trace("Not for us: %" PRIu32" != %" PRIu32,
-                      msg->host.id, local_nodeid);
+            crm_trace("Ignoring CPG message from ID %" PRIu32 " PID %" PRIu32
+                      ": for ID %" PRIu32" not %" PRIu32,
```

Comment:
normally would have a space after `PRIu32`; wasn't gonna say anything, but there is an indentation comment above

Reply:
thx